docs/profiling.md

```diff
@@ -3,6 +3,130 @@
 To profile JAX programs, there are currently two options: `nvprof` and XLA's
```

Comment:
Maybe say three?

Reply:
Good catch. I'm just gonna get rid of this, people can look at the ToC on the side :)